simtools/utils/general.py

```diff
@@ -348,7 +349,14 @@ def collect_data_from_http_yaml(url):
     try:
         with tempfile.NamedTemporaryFile() as tmp_file:
             urllib.request.urlretrieve(url, tmp_file.name)
-            data = yaml.load(tmp_file)
+            if url.endswith("yml") or url.endswith("yaml"):
```

Comment:
general comment for the function: I suggest changing the name of the function to `collect_data_file_from_http` and adding to the docstring the allowed types. This would reflect the fact that the function also allows getting json file. same comment for the `collect_data_from_yaml_or_dict`, which could change to `collect_data_from_file_or_dict`

Reply:
I agree, but I suggest to leave it for now as I see a very similar function in general.py (collect_dict_from_file). I think we can remove this one and make the code simpler - let me do this in a separate PR (I have a branch already, so no need to start an issue)

Reply:
Ok!